### PR TITLE
Add submitted status to application

### LINKED ISSUE #2381: Help IG team to identify when applications have been completed and submitted
Request from Amir. NOTE: Catherine needs to report back to Amir on decision

Amir and Liam are struggling to identify applications that have been finished and submitted amongst all those that have been partially completed but not yet submitted. Is there a way to make this easier for them? 

Their suggestion: Implement a completed button on the application form, to gage when applicant has officially submitted their application in full

### PR DESCRIPTION
**Pull request by lucyb**
When an application is submitted at the confirmation page it is now marked as submitted. This allows it to show up in the staff area list of applications separately to ongoing applications and will make it easier for reviewers to know which applications they need to look at.

Fixes #2381

Before

Screenshot from 2022-12-13 14-41-50

After

Screenshot from 2022-12-13 14-41-33

Screenshot from 2022-12-13 15-31-50

### CODE REVIEW

File: applications/migrations/0010_application_submitted_at_application_submitted_by_and_more.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    ),
+                    models.Q(
+                        ("submitted_at__isnull", False), ("submitted_by__isnull", False)
+                    ),
+                    _connector="OR",
+                ),
+                name="applications_application_both_submitted_at_and_submitted_by_set",
+            ),
+        ),
+    ]
```
Comment (ghickman) [inline applications/migrations/0010_application_submitted_at_application_submitted_by_and_more.py:61]: Can you give this a more meaningful name? Maybe something like `added_application_submitted_fields`, it really helps when you're looking at the list of migrations and trying to work out what they do.

**Review by ghickman — APPROVED** [review]

File: applications/models.py [hunk 2]
```diff
@@ -18,9 +18,16 @@ class Statuses(models.TextChoices):
         APPROVED_FULLY = "approved_fully", "Approved Fully"
         APPROVED_SUBJECT_TO = "approved_subject_to", "Approved Subject To"
         COMPLETED = "completed", "Completed"
+        SUBMITTED = "submitted", "Submitted"
         ONGOING = "ongoing", "Ongoing"
         REJECTED = "rejected", "Rejected"
 
+    submitted_by = models.ForeignKey(
+        "jobserver.User",
+        on_delete=models.CASCADE,
```
[the remark this answers is not in the document]
Comment (lucyb) [reply applications/models.py]: Should the other fields be set to PROTECT as well then?

File: applications/migrations/0010_application_submitted_at_application_submitted_by_and_more.py (hunk 1 above)
Comment (lucyb) [reply applications/migrations/0010_application_submitted_at_application_submitted_by_and_more.py]: Done. Although, we should probably look at squashing them fairly soon.

File: applications/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ghickman) [reply applications/models.py]: Good catch, yes!  The project field should stay as SET_NULL as that's a valid state for that field.

**Pull request merged**